File: LeetCode/Python/Problem Solving/Easy/Linked Lists/reverse.py
#Manuel Ortiz at 2022
#Extracted from: https://leetcode.com/problems/reverse-linked-list/

from node import Node
    
def reverse(head):
    actual = head
    prev = None
    while actual:
        actual.next, prev, actual = prev, actual, actual.next    
    return prev

if __name__ == '__main__':
    llist = head = Node(1)
    llist.append(2)
    llist.append(3)
    llist.display()
    rllist = rhead = reverse(head)
    rllist.display()
    

File: LeetCode/JavaScript/Easy/Linked Lists/reverse.js
import { Node } from './node.js';

const reverseList = head => {
    let currentNode = head;
    let prev = null;
    
    while (currentNode != null){
        [currentNode.next, prev, currentNode] = [prev, currentNode, currentNode.next];
    }
    
    return prev;
    
};

let head = new Node(5)
head.append(4,3,2,1)
head.display();
head = reverseList(head);
head.display();